autopilot/contractor.go

```diff
@@ -33,16 +45,64 @@ func newContractor(ap *Autopilot) *contractor {
 
 func (c *contractor) applyConsensusState(cfg Config, state bus.ConsensusState) {
 	c.blockHeight = state.BlockHeight
-	// TODO: update current period
+	if c.blockHeight > c.currentPeriod+cfg.Contracts.Period {
+		c.currentPeriod += cfg.Contracts.Period
+	}
 }
 
-func (c *contractor) remainingFunds(cfg Config) (remaining types.Currency) {
-	var spent types.Currency // TODO: period spending
+func (c *contractor) contractSpending(id types.FileContractID) (bus.ContractSpending, error) {
+	// fetch contract chain
+	contracts, err := c.ap.bus.ContractHistory(id)
+	if err != nil {
+		return bus.ContractSpending{}, err
+	}
 
-	if cfg.Contracts.Allowance.Cmp(spent) > 0 {
-		remaining = cfg.Contracts.Allowance.Sub(spent)
+	// compute total spending
+	var total bus.ContractSpending
+	for _, contract := range contracts {
+		spending, err := c.ap.bus.ContractSpending(contract.ID)
+		if err != nil {
+			return bus.ContractSpending{}, err
+		}
+		total = total.Add(spending)
+	}
+
+	return total, nil
+}
+
+func (c *contractor) currentPeriodSpending() (types.Currency, error) {
```

Comment:
currently is kind of annoying we have `bus.Contract` and `rhpv2.Contract` - perhaps we should consider renaming `bus.Contract` to `ContractHeader` or `ContractInfo` so we can call an actual contract a contract.